devtools/gradle/src/main/java/io/quarkus/gradle/ConnectorDependencyResolver.java

```diff
@@ -3,38 +3,37 @@
 import java.nio.file.Path;
 import java.util.Collections;
 import java.util.List;
-import java.util.Objects;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import org.apache.maven.model.Dependency;
 import org.gradle.tooling.BuildException;
-import org.gradle.tooling.GradleConnector;
-import org.gradle.tooling.ProjectConnection;
-import org.gradle.tooling.model.eclipse.EclipseExternalDependency;
-import org.gradle.tooling.model.eclipse.EclipseProject;
+
+import io.quarkus.bootstrap.resolver.QuarkusGradleModelFactory;
+import io.quarkus.bootstrap.resolver.model.QuarkusModel;
+import io.quarkus.runtime.LaunchMode;
 
 public final class ConnectorDependencyResolver {
 
     private List<Dependency> dependencies = null;
```

Comment:
@mgorniew why are we using `org.apache.maven.model.Dependency` here?

Reply:
This is model used by `io.quarkus.devtools.project.buildfile.BuildFile`.

Reply:
@mgorniew I think we can easily change that :) just remove `getDependencies` and have `getDependenciesKeys` abstract, and move the current `getDependenciesKeys` logic to the `MavenBuildFile`, then you can avoid this dependency in Gradle..

Reply:
`AppArtifactKey` doesn't contain version and there is `getInstalled` method which uses `getDependencies` to create `AppArtifactCoords`.

Reply:
Then use `AppArtifactCoords` isnted of keys :)

Reply:
The issue is that we are using Maven's model Dependency in QuarkusPlatformDescriptor. So either way, we are going to be converting unless we change it in QuarkusPlatformDescriptor and everywhere else.

Reply:
It could put a start to it though.

Reply:
Sure, I will change maven `Dependency` to `AppArtifactCoords`.